Sources/UI/TruncatingLabel/TruncatingLabel.swift

```diff
@@ -0,0 +1,325 @@
+//
+//  TruncatingLabel.swift
+//  PovioKit
+//
+//  Created by Toni K. Turk on 25/08/2022.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import UIKit
+
+// 1. Both strings fit into a single line
+//  ---------------------------------------------
+// |   This is a short text.   secondary label   |
+//  ---------------------------------------------
+//
+// 2. Strings do not fit into a single line, however,
+// there is enough space for both of them in two lines.
+//  ---------------------------------------------
+// |   This is a longer single-line text.        |
+// |                           secondary label   |
+//  ---------------------------------------------
+//
+//  ---------------------------------------------
+// |   This is a longer two-line text, lorem     |
+// |   ipsum dolor.            secondary label   |
+//  ---------------------------------------------
+//
+// 3. Strings do not fit into a single line and there is also
+// not enough space for the primary string in the second line.
+// Primary string, therefore, has to be truncated with '...'.
+//  ---------------------------------------------
+// |   This is a long text that has to be split  |
+// |   into two lines, ho...    secondary label  |
+//  ---------------------------------------------
+//
+// 4. There is not enough space for the secondary text
+// in the whole line. Truncate primary text in the first line.
+//  ---------------------------------------------
+// |   This is a long text that has to be sp...  |
+// |   secondary label is too long to be disp... |
+//  ---------------------------------------------
+//
+public class TruncatingLabel: UIView {
+  // @NOTE: - For autolayout to work properly, at least the width of the view
+  // should be 'statically' determined (that is, layoutSubviews() should
+  // receive a frame with valid width) by the constraint setup, i.e., bind
+  // the leading and trailing constraints to the parent view. The height of the
+  // view is dynamic and the intrinsic content size updated accordingly.
+  
+  private var primaryTextAttributes: [NSAttributedString.Key: Any] = [
+    .font: UIFont.systemFont(ofSize: 14),
+    .foregroundColor: UIColor.gray
+  ]
+  private var secondaryTextAttributes: [NSAttributedString.Key: Any] = [
+    .font: UIFont.systemFont(ofSize: 12),
+    .foregroundColor: UIColor.blue
+  ]
+  
+  public var primaryColor: UIColor {
+    get {
+      primaryTextAttributes[.foregroundColor]! as! UIColor
```

Comment:
Although we are sure atm, the values are non-nil, this could change in the future. I would change this property to return an optional value instead. Just for the sake of safety. What do you think?